File: classes/MovementQueue.py
class MovementQueue:
	def __init__(self):
		self.moveList = []
		self.index = 0

	def next(self):
		move = self.moveList[self.index]
		if self.index + 1 == len(self.moveList):
			self.index = 0
		else:
			self.index += 1
		return move

	def insertIntoQueue(self, newMove, position):
		self.moveList = self.moveList[:position] + [newMove] + self.moveList[position:]
		if position < self.index:
			self.index += 1

File: Assets/Classes/Characters/MovementQueue.cs
using System.Collections.Generic;

public class MovementQueue {
	public List<Movement> moveList;
	private int index;

	public MovementQueue() {
		moveList = new List<Movement> ();
		index = 0;
	}

	public Movement next() {
		Movement move = moveList [index];
		incrementIndex ();
		return move;
	}

	private void incrementIndex() {
		if (index + 1 == moveList.Count) {
			index = 0;
		} else {
			index++;
		}
	}

	public void addToQueue(Movement move) {
		insertIntoQueue (move, moveList.Count);
	}

	public void insertIntoQueue(Movement move, int position) {
		moveList.Insert (position, move);
		if (position < index) {
			index++;
		}
	}
}
